### PR TITLE
TEC-1063 [FE] Group flytile and title into one heading

### PR DESCRIPTION
**Pull request by yasgreen93**
https://jira.economist.com/browse/TEC-1063
feat: grouped the flytitle and title into one h2 and corrected the css to have the same styling
Needs merging to be able to update fe-blogs.

### CODE REVIEW

**Comment by fabiosantoscode** [comment]
I'd expect parts/title to be deleted here. Also, don't you need to change the heading level of teasers eventually? This is where you do the change. The `FlyTitle` component and the BlogPost component start taking a `HeadingComponent` prop, defaulting to `'h2'`. Then instead of `<h2 ....>stuff</h2>` you just do `<HeadingComponent ...>stuff</HeadingComponent>` and you have a customisable header. It can be an `h2`, `h3`, or even a div!

**Comment by fabiosantoscode** [comment]
I'm also wondering if the remaining component should be called `FlyTitle`, since it's now a title and a flytitle, and the most important of the two is the Title. But I'm not the person to go to for naming.

**Comment by dmolin** [comment]
I don't even understand what 'FlyTitle' means LOL

File: src/index.css [hunk 1]
```diff
@@ -48,15 +49,17 @@
   line-height: var(--text-line-height-sans-light-on-step--2);
 }
 
-.blog-post__flytitle {
+.main-content__blog-post .flytitle-and-title__flytitle {
```
Comment (fabiosantoscode) [inline src/index.css]: We actually want these styles in fe-blogs!

File: src/index.js [hunk 2]
```diff
@@ -29,6 +27,7 @@ export default class BlogPost extends React.Component {
       sectionUrl: React.PropTypes.string,
       flyTitle: React.PropTypes.string,
       title: React.PropTypes.string.isRequired,
+      TitleComponent: React.PropTypes.oneOfType([ React.PropTypes.string, React.PropTypes.func ]).isRequired,
```
Comment (fabiosantoscode) [inline src/index.js]: Sorry, I know I wrote this, but TitleComponent can't be a string. Do you know why?

**Comment by fabiosantoscode** [comment]
Tests are failing, do they need updating? Some may be referring to deleted or modified code

**Comment by dmolin** [comment]
LGTM :shipit:

**Pull request merged**